metrics/sentry/cron.py

```diff
@@ -0,0 +1,59 @@
+import os
+
+import sentry_sdk
+from sentry_sdk.crons import capture_checkin
+from sentry_sdk.crons.consts import MonitorStatus
+
+
+class Cron:
+    def __init__(self):
+        sentry_sdk.init(
+            dsn=os.environ.get("SENTRY_DSN"),
+        )
+
+    _monitor_config = {
+        "schedule": {"type": "crontab", "value": "@daily"},
+        "timezone": "Etc/UTC",
+        # If an expected check-in doesn't come in `checkin_margin`
+        # minutes, it'll be considered missed
+        "checkin_margin": 30,
+        # The check-in is allowed to run for `max_runtime` minutes
+        # before it's considered failed
+        "max_runtime": 10,
+        # It'll take `failure_issue_threshold` consecutive failed
+        # check-ins to create an issue
+        "failure_issue_threshold": 1,
+        # It'll take `recovery_threshold` OK check-ins to resolve
+        # an issue
+        "recovery_threshold": 1,
+    }
+
+    def get_monitor(self, modname):
+        monitor_slug = f"metrics-{modname}"
+        return self.Monitor(monitor_slug, self._monitor_config)
+
+    class Monitor:
+        def __init__(self, monitor_slug, monitor_config):
+            self.monitor_slug = monitor_slug
+            self.monitor_config = monitor_config
+
+        check_in_id = None
```

Comment:
I'm a bit confused about what `check_in_id` will be set to for instances of Monitor after the first task is run. As I understand it, `check_in_id` is a static variable because it's defined in the class, but since it's a class within a class it becomes a bit complicated. Would it make more sense to define it as an instance variable instead or have I missed something?

Reply:
It should be an instance variable - this is something that regularly catches me out in Python (first learned OO with C# and Java)! Thanks for the catch